docs/pages/database-access/guides/redis.mdx

```diff
@@ -0,0 +1,350 @@
+    $ tctl auth sign --format=redis --host=redis1.example.com,10.0.0.1 --out=redis1 --ttl=2190h
+    $ tctl auth sign --format=redis --host=redis2.example.com,10.0.0.2 --out=redis2 --ttl=2190h
+    ```
+
+    (!docs/pages/includes/database-access/ttl-note.mdx!)
+
+    The command will create three files: `server.cas` with Teleport's certificate
+    authority, `server.key` with generated private key and `server.crt` with generated user certificate. You will
+    need these files to enable mutual TLS on your Redis server.
+  </TabItem>
+</Tabs>
+
+(!docs/pages/includes/database-access/rotation-note.mdx!)
+
+Use the generated secrets to enable mutual TLS in your `redis.conf` configuration
+file and restart the database:
+
+<Tabs>
+  <TabItem label="Standalone">
+    ```ini
+    tls-port 6379
+    port 0
+
+    aclfile /path/to/users.acl
+
+    tls-ca-cert-file /path/to/server.cas
+    tls-cert-file /path/to/server.crt
+    tls-key-file /path/to/server.key
+    tls-protocols "TLSv1.2 TLSv1.3"
+    ```
+  </TabItem>
+  <TabItem label="Cluster">
+    ```ini
+    tls-port 7001
+    port 0
+
+    cluster-enabled yes
+
+    tls-replication yes
+    tls-cluster yes
+
+    aclfile /path/to/users.acl
+
+    masterauth GENERATED_STRONG_PASSWORD
+    masteruser replica-user
+
+    tls-cert-file /usr/local/etc/redis/certs/server.crt
+    tls-key-file /usr/local/etc/redis/certs/server.key
+    tls-ca-cert-file /usr/local/etc/redis/certs/server.cas
+    tls-protocols "TLSv1.2 TLSv1.3"
+    ```
+  </TabItem>
+</Tabs>
+
+Once mutual TLS has been enabled, you will no longer be able to connect to
+the cluster without providing a valid client certificate. You can use
+`tls-auth-clients optional` setting to allow connections
+from clients that do not present a certificate.
+
+See [Configure TLS/SSL](https://redis.io/topics/encryption)
+in Redis documentation for more details.
+
+<Admonition type="note" title="Create Redis Cluster">
+  When you're configuring Redis Cluster you need to create the cluster using `redis-cli`
+
+  `redis-cli --cluster create` accepts only IP addresses, so the
+
+  Use below command to create the cluster:
+
+  ```shell
+  export REDISCLI_AUTH=STRONG_GENERATED_PASSWORD
+  export CERTS_DIR=/path/to/certs/
+
+  redis-cli --user alice --cluster-replicas 1 --tls --cluster-yes \
+  --cluster create ${IP1}:7001 ${IP2}:7002 ${IP3}:7003 ${IP4}:7004 ${IP5}:7005 ${IP6}:7006 \
+  --cacert ${CERTS_DIR}/server.cas --key ${CERTS_DIR}/server.key --cert ${CERTS_DIR}/server.crt
+  ```
+
+</Admonition>
+
+<Admonition type="note" title="Redis Cluster Only">
+
+  To enable Redis cluster mode in Teleport add `mode=cluster` parameter to connection uri in teleport config file.
+
+  ```yaml
+  databases:
+    - name: "redis-cluster"
+      uri: "rediss://redis.example.com:6379?mode=cluster"
+  ```
+
+</Admonition>
+
+## Step 5/5. Connect
+
+Log into your Teleport cluster and see available databases:
+
+```code
+$ tsh login --proxy=teleport.example.com --user=alice
```

Comment:
Let's make this a Tabs component for "Self-Hosted" and "Teleport Cloud," where for Cloud, `--proxy` is set to `mytenant.teleport.sh`.

Reply:
Done.